jleeboyd/family-bank-app
# Refactor delete buttons

### User Story
#69 - Refactor delete transaction
#67 - Create delete account
#58 - Refactor tests and >80%

This PR: 
- Migrates the delete transaction button inside the transaction activity
- Adds in a new delete account button in the account activity
- Yes/No Dialogs are now implemented for deleting
- Refactors tests and improves coverage to ~90%
- Changes the app name and icon

**Testing steps below screenshots**

### Screenshots:

**Delete icon in account activity**
![Screen Shot 2020-11-29 at 8 16 17 PM](https://user-images.githubusercontent.com/52220118/100568344-da526380-327f-11eb-8fa3-4bcae7547d79.png)

**Delete icon in transaction activity**
![Screen Shot 2020-11-29 at 8 16 24 PM](https://user-images.githubusercontent.com/52220118/100568346-db839080-327f-11eb-90c8-f96b0d20b1c0.png)

**Delete dialog**
![Screen Shot 2020-11-29 at 8 16 42 PM](https://user-images.githubusercontent.com/52220118/100568350-dcb4bd80-327f-11eb-994e-2b4064b85c1a.png)

**New app name and icon**
![Screen Shot 2020-11-29 at 8 15 13 PM](https://user-images.githubusercontent.com/52220118/100568263-9f503000-327f-11eb-81ef-3098c0523441.png)

### Context / Additional Information

**To Test:** 
- Create an account 
- Create a new transaction for that account
- Click the newly created transaction and check that the proper status is displayed
- Now delete the transaction
   - Click "YES"
- [ ] See that the transaction is crossed out, its status is "Deleted", and a toast displays
- Now delete the account
   - Click "YES"
- [ ] See that the account and all transactions are deleted from the database and a toast displays

### Acceptance Criteria
- [x] - Delete button in transaction activity
- [x] - Delete button in account activity
- [x] - Status is shown in the transaction activity
- [x] - Deleting account deletes all transaction data
- [x] - Deleting transaction crosses out specific transaction
- [x] - Dialog confirms for both
